File: 3791-fruits-into-baskets-iii/fruits-into-baskets-iii.java
class Solution {
    public int numOfUnplacedFruits(int[] fruits, int[] baskets) {
        int unp = 0;
        int max = 0;
        List<Integer> list = new ArrayList<>();
        for(int j = 0 ; j < baskets.length ;j++){
            list.add(baskets[j]);
            max = Math.max(max, baskets[j]);
        }
        for(int i = 0 ; i < fruits.length ; i++){
            if(fruits[i] > max){
                unp++;
                continue;
            }
            boolean placed = false;
            for(int j = 0 ; j < list.size() ; j++){
                if(list.get(j) >= fruits[i]){
                    list.remove(j);
                    placed = true;
                    break;
                }
            }
            if(!placed) unp++;
        }
        return unp;
    }
}

File: 3791-fruits-into-baskets-iii/fruits-into-baskets-iii.cpp
class Solution {
public:
    int numOfUnplacedFruits(vector<int>& fruits, vector<int>& baskets) {
        vector<int> basket_copy;
        int count = 0;
        int maxi = 0;
        for(int i:baskets){
            basket_copy.push_back(i);
            maxi = max(maxi,i);
        }
        for(int i =0;i < fruits.size();i++){
            int fruit = fruits[i];
            if(fruit > maxi){
                count++;
                continue;
            }
            bool is_placed = false;
            for(int j = 0;j < basket_copy.size();j++){
                int basket_size = basket_copy[j];
                if(basket_size >= fruit){
                    is_placed = true;
                    basket_copy.erase(basket_copy.begin() + j);
                    break;
                }
            }
            if(!is_placed) count++;
        }
        return count;
    }
};